src/components/VideoProvider/AttachVisibilityHandler/AttachVisibilityHandler.tsx

```diff
@@ -0,0 +1,44 @@
+import { isMobile } from '../../../utils';
+import { useEffect, useRef } from 'react';
+import useLocalVideoToggle from '../../../hooks/useLocalVideoToggle/useLocalVideoToggle';
+import useVideoContext from '../../../hooks/useVideoContext/useVideoContext';
+
+/* 
+  This component adds a visibilitychange handler to the document when
+  the user is using a mobile device. When the handler detects that
+  the browser has been backgrounded, it unpublishes the users local
+  video track. The browser cannot send video to the room when it has
+  been backgrounded, so unpublishing the track stops video capture
+  on the device, and triggers a UI update for all other participants
+  to show that this user's video track has been turned off.
+*/
+
+export default function AttachVisibilityHandler() {
+  const { room } = useVideoContext();
+  const [isVideoEnabled, toggleVideoEnabled] = useLocalVideoToggle();
+  const shouldRepublishVideoOnForeground = useRef(false);
+
+  useEffect(() => {
+    if (isMobile) {
```

Comment:
I try to stay away from browser detection if possible and use feature detection instead. Should we do that here?

Reply:
That's a good point. For this code, I based it off of @manjeshbhargav's changes in the same quickstart feature: https://github.com/twilio/video-quickstart-js/blob/mobile-browser-support/quickstart/src/joinroom.js#L271

Is there something that you have in mind for feature detection?

Reply:
In general, it depends on the feature we're using. In this case, it seems like we can remove the `isMobile` check. If the browser doesn't support `visibilitychange`, this code will not do anything, is that correct?

Reply:
`visibilitychange` is supported on mobile and desktop browsers, but we only want this feature on mobile. We want to unpublish the user's video when a mobile browser is backgrounded, but we don't want to unpublish the user's video when the user changes tabs in a desktop browser. 

[This article](https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent) recommends feature detection over user-agent sniffing, but in this case I think that user-agent sniffing should be sufficient. Although now I'm wondering if we should also check for [Firefox tablets](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent/Firefox#Mobile_and_Tablet_indicators) as well.

I'll ask in the quickstart PR.

Reply:
> but we don't want to unpublish the user's video when the user changes tabs in a desktop browser.

Please help me understand why? Because this sounds like an improvement.

Reply:
Users are often multi-tasking when using the desktop app. If you switch to another tab, do you want your local video to be unpublished from the room? Other users could only see you when you have the app in the active tab. There would be no way for users to see if you if you are using another tab. 

On mobile, when the app is backgrounded, the camera stops sending frames. So we unpublish the video track to 1) update the UI in other participant's apps to show that the mobile user's video is gone (this will display the correct icon), and 2) free up the camera resource on the mobile device when it is not in use. 

On desktop, the camera still functions perfectly fine when the user changes tabs, so there is no need to unpublish it.

Reply:
Ok. I understand the use case.

Reply:
Sorry, I probably should have referenced the recently published best practices guide that this PR is implementing: https://www.twilio.com/docs/video/build-js-video-application-recommendations-and-best-practices#application-backgrounding-in-mobile-browsers

Note: this guide will be amended to recommend that video tracks are unpublished, not disabled.